src/Context/AuthContext.tsx

```diff
@@ -30,7 +44,7 @@ export const AuthContextProvider: React.FC<{ children: ReactNode }> = ({
 }) => {
   const [adminData, setAdminData] = useState<DecodedToken | null>(null);
   const [userRole, setUserRole] = useState<string | null>(null);
-
+  const [currentUser, setCurrentUser] = useState<currentUser | null>(null);
   const Token = localStorage.getItem("token");
 
   const saveAdminData = () => {
```

Comment:
This function is unnecessary, as we have all needed information in the getCurrentUser